### PR TITLE
add flush index UI to index operation

### PR DESCRIPTION
**Pull request by zhichao-aws**
### Description
add flush index UI to index operation

### Issues Resolved
https://github.com/opensearch-project/index-management-dashboards-plugin/issues/700

### Check List
- [ ] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by zhichao-aws** [comment]
E2E tests workflow is blocked by build error in [IM backend plugin #762](https://github.com/opensearch-project/index-management/issues/762). Can pass them in local environment.

File: public/containers/FlushIndexModal/FlushIndexModal.tsx [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+    const indexPayload = unBlockedItems.join(",");
+    const result = await services.commonService.apiCaller({
+      endpoint: "indices.flush",
+      data: {
+        index: indexPayload,
+      },
+    });
+    if (result && result.ok) {
+      const flushedItems = flushAll ? "all open indexes" : `[${indexPayload}]`;
+      coreServices.notifications.toasts.addSuccess(`Flush ${flushedItems} successfully`);
```
Comment (SuZhou-Joe) [inline public/containers/FlushIndexModal/FlushIndexModal.tsx:73]: join(", "), add additional space to make the message truncatable.

File: public/containers/FlushIndexModal/FlushIndexModal.tsx [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+  }, [visible, flushTarget, selectedItems]);
+
+  if (!visible) {
+    return null;
+  }
+
+  return (
+    <EuiModal onClose={onClose}>
+      <EuiModalHeader>
+        <EuiModalHeaderTitle data-test-subj="Flush Modal Title">Flush {flushTarget}</EuiModalHeaderTitle>
```
Comment (SuZhou-Joe) [inline public/containers/FlushIndexModal/FlushIndexModal.tsx:138]: Recommend using camel case for data-test-subj: https://github.com/opensearch-project/OpenSearch-Dashboards/blob/main/DEVELOPER_GUIDE.md#camel-case-id-and-data-test-subj

File: public/containers/FlushIndexModal/FlushIndexModal.tsx [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+      </EuiModalHeader>
+
+      <EuiModalBody>
+        <div style={{ lineHeight: 1.5 }}>
+          {/* we will not display this part if not flushAll and there is no flushable items */}
+          {flushAll && (
+            <>
```
Comment (SuZhou-Joe) [inline public/containers/FlushIndexModal/FlushIndexModal.tsx:145]: No need to wrap addtional <></> in this line I suppose.

File: public/containers/FlushIndexModal/FlushIndexModal.tsx [hunk 4]
```diff
@@ -0,0 +1,182 @@
...
+              <p>{messageMap[flushTarget]}</p>
+              <ul style={{ listStyleType: "disc", listStylePosition: "inside" }}>
+                {unBlockedItems.map((item) => (
+                  <li key={item}>{item}</li>
+                ))}
+              </ul>
+            </>
+          )}
+          <EuiSpacer />
+          <EuiCallOut data-test-subj="Flush Blocked Callout" color="warning" hidden={!blockedItems.length}>
```
Comment (SuZhou-Joe) [inline public/containers/FlushIndexModal/FlushIndexModal.tsx:160]: camel case

File: public/containers/FlushIndexModal/FlushIndexModal.tsx [hunk 5]
```diff
@@ -0,0 +1,182 @@
...
+                <li key={item}>{item}</li>
+              ))}
+            </ul>
+          </EuiCallOut>
+          <EuiSpacer />
+        </div>
+      </EuiModalBody>
+
+      <EuiModalFooter>
+        <EuiButtonEmpty data-test-subj="Flush Cancel Button" onClick={onClose}>
```
Comment (SuZhou-Joe) [inline public/containers/FlushIndexModal/FlushIndexModal.tsx:173]: camel case

File: public/containers/FlushIndexModal/FlushIndexModal.tsx [hunk 6]
```diff
@@ -0,0 +1,182 @@
...
+import { CoreStart } from "opensearch-dashboards/public";
+import { CoreServicesContext } from "../../components/core_services";
+import { getErrorMessage } from "../../utils/helpers";
+import { ServicesContext } from "../../services";
+import { indexBlockedPredicate, aliasBlockedPredicate, dataStreamBlockedPredicate, filterBlockedItems } from "../../utils/helpers";
+import { IndexOpBlocksType } from "../../utils/constants";
+import { CatIndex, DataStream } from "../../../server/models/interfaces";
+import { IAlias } from "../../pages/Aliases/interface";
+
+type FlushTarget = "indices" | "data stream" | "alias";
```
Comment (SuZhou-Joe) [inline public/containers/FlushIndexModal/FlushIndexModal.tsx:27]: An enum will be better in this case.

File: public/containers/FlushIndexModal/FlushIndexModal.tsx (hunk 3 above)
Comment (zhichao-aws) [reply public/containers/FlushIndexModal/FlushIndexModal.tsx:145]: will change it

File: public/containers/FlushIndexModal/FlushIndexModal.tsx (hunk 1 above)
Comment (zhichao-aws) [reply public/containers/FlushIndexModal/FlushIndexModal.tsx:70]: will change it

**Review by xluo-aws — APPROVED** [review]

**Review by xluo-aws — DISMISSED** [review]

**Review by SuZhou-Joe — APPROVED** [review]

**Review by Hailong-am — APPROVED** [review]

**Review by SuZhou-Joe — APPROVED** [review]

**Pull request merged**